lib/vdc/DCMPAS.cpp

```diff
@@ -1900,3 +1932,212 @@ bool DCMPAS::DerivedCoordVertFromCell::VariableExists(
 
 	return( _dc->VariableExists(ts, _inName, -1, -1));
 }
+
+//////////////////////////////////////////////////////////////////////
+//
+// Class definitions for derived data variables
+//
+//////////////////////////////////////////////////////////////////////
+
+
+
+DCMPAS::DerivedZonalMeridonal::DerivedZonalMeridonal(
+	string derivedVarName, DC *dc, NetCDFCollection *ncdfc,
+	string normalVarName, string tangentialVarName, bool zonalFlag
+
+) : DerivedDataVar(
+	derivedVarName
+) {
+	_dc = dc;
+	_ncdfc = ncdfc;
+	_normalVarName = normalVarName;
+	_tangentialVarName = tangentialVarName;
```

Comment:
```suggestion
	_tangentialVarName = std::move(tangentialVarName); // pass-by-value and move 
```

The same applies to variable `_normalVarName`

Reply:
_normalVarName is a string, not a raw pointer. So I don't understand the comment.

Reply:
"pass-by-value-and-move" is a programming idiom that avoids unnecessary memory allocations. It's especially useful when the parameters passed around have dynamically allocated memory space, like string in this case. Please see this document for more detail:
https://paper.dropbox.com/doc/Code-Snippet--Az4I~xnjxteyg~vhPGZWTU21Ag-H2oXdHWnNj5b2c4aMovCh

Reply:
What are the practical benefits of pass-by-value-and-move for small strings and a function that will only be called a handful of times?

Reply:
Well, I see it very powerful just because of its simplicity -- you do it in this fashion **whenever** you receive some complex object from the parameter and make a copy by yourself.

- Do you need to evaluate if the string/vector is a big one or small one? No. 
- Do you need to evaluate if this function is called repeatedly or occasionally? No. 
- Do you need to evaluate if the parameter is a const type or not? No
- Do you need to evaluate if you should pass by value or by reference? No

Plus, it's really not big difference in terms of typing. Why not?

Reply:
To be clear, it guarantees the least number of dynamic memory allocations in every case without the need of the programmer to choose between design options such as values/references, const/non-const, etc.

Reply:
I understand that it is more efficient in terms of memory space and maybe even computation, but this is not a place where any of that matters. I don't really care about making the change here, but I don't want to start down the road where every line of C++ code is scrutinized for optimal performance. We should focus our performance tuning efforts where memory usage or computational efficiency really matters. Where these considerations are not important we should emphasize readability, simplicity, robustness, and maintainability. Do we need to, for example, worry about the use of postfix increment operators (x++)? Prefix operators (++x) are more efficient, and should be used when looping over millions of elements (such as when using the Grid iterators), but are we going to require prefix operators for looping over n==3 elements because it saves a few nanoseconds :-). I hope not.

Reply:
Hmm, I think your argument applies better to a scenario where option A and B both have some pros and cons, but in inconsequential cases, choose whatever you like instead of an optimal one. I agree with this philosophy.

In my view, the current scenario is that option A is superior to option B, all the time. As a result, I'm advocating that we always use option A and forget there's even an option B  there. Pass-by-value-and-move is option A in this scenario.

I'm advocating this practice because I often find making decisions from available choices imposes a mental burden on me, and I'm always happy to eliminate available choices when writing code. As a team, if all code follows this programming idiom, then even if another function comes to the reviewer and doesn't show any obvious pattern on its frequency being called and average size of objects being passed around, the reviewer can still be assured that there's no possible improvement. But afterall if you feel there are other reasons to keep it the way it currently works, I'm happy to mark this conversation resolved.

Reply:
Let's move ahead without spending too much time on this issue :)